src/Store/FileExplorerStore.re

```diff
@@ -149,7 +149,17 @@ let start = () => {
       | {active, _} when active != maybeFilePath =>
         let state = setActive(maybeFilePath, state);
         switch (maybeFilePath) {
-        | Some(path) => revealAndFocusPath(path, state)
+        | Some(path) =>
+          let autoReveal =
+            Oni_Core.Configuration.getValue(
+              c => c.explorerAutoReveal,
+              state.configuration,
+            )
+            switch (autoReveal) {
+            | `HighlightAndScroll => revealAndFocusPath(path, state)
+            | `HighlightOnly => (setFocus(Some(path), state), Isolinear.Effect.none)
+            | `NoReveal => (state, Isolinear.Effect.none)
```

Comment:
@bryphe This is still causing files to be highlighted (but not scrolled through) with the false setting/`NoReveal` type. Any idea why that might be? Everything else seems to work as expected.

Reply:
Ah, was just looking at this - I was testing out your branch, and I saw the same behavior as you (if I set `explorer.autoReveal` to `false` - the `NoReveal` is set, but we still get a highlight).

I think it's related to this line: https://github.com/onivim/oni2/blob/46bde3b1820efee54e5d77884e662962357c7b7c/src/Store/FileExplorerStore.re#L150

Confusingly - there is a concept of 'active' _and_ 'focus'. I think we may need to refactor those to a single concept - I'm not sure why we need them both, or what case active != focus. In the meantime - if we move that `setActive` call into the `HighlightScroll`/`HighlightOnly` cases, it seems to work as expected.

Reply:
I believe "active" is the file in the currently active editor and "focus" is the manually selected one. If you unify them, you'd have to change the active editor when you navigate up and down the tree using the keyboard, for example. Or you could of course not specifically highlight the active editor, but just change the focus when the active editor changes. FWIW, VS Code makes this same distinction.